app/src/main/java/org/oppia/app/options/OptionsFragment.kt

```diff
@@ -0,0 +1,185 @@
+          ), 1
+        )
+        return true
+      }
+    }
+
+    val appLanguagePref = findPreference<Preference>(getString(R.string.key_app_language))
+    appLanguagePref.summary = appLanguage
+    appLanguagePref.onPreferenceClickListener = object : Preference.OnPreferenceClickListener {
+      override fun onPreferenceClick(preference: Preference): Boolean {
+        startActivityForResult(
+          AppLanguageActivity.createAppLanguageActivityIntent(
+            requireContext(),
+            getString(R.string.key_app_language),
+            appLanguagePref.summary.toString()
+          ), 2
+        )
+        return true
+      }
+    }
+
+    val defaultAudioPref = findPreference<Preference>(getString(R.string.key_default_audio))
+    defaultAudioPref.summary = audioLanguage
+    defaultAudioPref.onPreferenceClickListener = object : Preference.OnPreferenceClickListener {
+      override fun onPreferenceClick(preference: Preference): Boolean {
+        startActivityForResult(
+          DefaultAudioActivity.createDefaultAudioActivityIntent(
+            requireContext(),
+            getString(R.string.key_default_audio),
+            defaultAudioPref.summary.toString()
+          ), 3
+        )
+        return true
+      }
+    }
+  }
+
+  private fun bindPreferenceSummaryToValue(typeOfValue: String, preference: Preference) {
+    preference.onPreferenceChangeListener = bindPreferenceSummaryToValueListener
+
+    bindPreferenceSummaryToValueListener.onPreferenceChange(
+      preference, PreferenceManager
+        .getDefaultSharedPreferences(preference.context)
+        .getString(preference.key, typeOfValue)
+    )
+  }
+
+  private val bindPreferenceSummaryToValueListener =
+    Preference.OnPreferenceChangeListener { preference, newValue ->
+      val stringValue = newValue.toString()
+      when {
+        preference.key == getString(R.string.key_story_text_size) -> // update the changed Text size  to summary filed
+        {
+          preference.summary = stringValue
+          when (stringValue) {
+            "Small" -> {
+              profileManagementController.updateStoryTextSize(profileId, 16f)
+            }
+            "Medium" -> {
+              profileManagementController.updateStoryTextSize(profileId, 18f)
+            }
+            "Large" -> {
+              profileManagementController.updateStoryTextSize(profileId, 20f)
+            }
+            "Extra Large" -> {
+              profileManagementController.updateStoryTextSize(profileId, 22f)
+            }
+          }
+        }
+        preference.key == getString(R.string.key_app_language) -> // update the changed language  to summary filed
+        {
+          preference.summary = stringValue
+          profileManagementController.updateAppLanguage(profileId, stringValue)
+        }
+        preference.key == getString(R.string.key_default_audio) -> // update the changed audio language  to summary filed
```

Comment:
Prefer writing comments above this line. as such comments can easily get skipped.

Reply:
Also, write proper comment meaning the spacing between words should be single space, first character should be capital, full-stop to end the sentence.

Reply:
Corrected comment line.